jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/info/JvmInfo.java

```diff
@@ -0,0 +1,168 @@
+						props.put(keyString, valuePresent);
+						ArrayType<?> valueArrayType = (ArrayType<?>) valueType;
+						CompositeType elementType = getCompositeType(valueArrayType.getElementOpenType());
+						CompositeData valueArray[] = (CompositeData[]) value;
+						final int arrayLength = valueArray.length;
+						props.put(keyString + ARRAY_SIZE, arrayLength);
+						for (int i = 0; i < arrayLength; ++i) {
+							String newPrefix = keyString + '.' + Integer.toString(i) + '.';
+							addFields(props, newPrefix, elementType, valueArray[i]);
+						}
+					} else {
+						CompositeType newCompType = getCompositeType(valueType);	
+						String newPrefix = keyString + '.';
+						CompositeData newCompData = (CompositeData) value;
+						addFields(props, newPrefix, newCompType, newCompData);
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Get the CompositeType object based on a name
+	 * @param typeIdentifier Name of the type
+	 * @return CompositeType for that type; null if the type is unrecognized.
+	 */
+	protected static CompositeType getCompositeType(OpenType<?> typeIdentifier) {
+		CompositeType result = null;
+
+		String typeName = typeIdentifier.getTypeName();
+		switch (typeName) {
+		case "java.lang.management.LockInfo": //$NON-NLS-1$
+			result = LockInfoUtil.getCompositeType();
+			break;
+		case "java.lang.management.MonitorInfo": //$NON-NLS-1$
+			result = MonitorInfoUtil.getCompositeType();
+			break;
+		case "java.lang.StackTraceElement": //$NON-NLS-1$
+			result = StackTraceElementUtil.getCompositeType();
+			break;
+		default:
+			break;
+		}
+		return result;
+	}
+
+	/**
+	 * Convert a Properties into a CompositeData object
+	 * @param props Source of the data
+	 * @param propertyPrefix Used to filter the data
+	 * @param compType type description
+	 * @return Filled-in CompositeData object
+	 * @throws IOException if the properties are invalid
+	 */
+	protected static CompositeData extractFields(DiagnosticProperties props, 
+			String propertyPrefix, CompositeType compType) throws IOException {
+		Set<String> keys = compType.keySet();
+		final int numKeys = keys.size();
+		String[] itemNames = keys.toArray(new String[numKeys]);
+		Object itemValues[] = new Object[numKeys];
+		try {
+			for (int cursor = 0; cursor < numKeys; ++cursor) {
+				String k = itemNames[cursor];
+				String keyString = propertyPrefix + k;
+				Object value = null;
+				final OpenType<?> valueType = compType.getType(k);
+				if (JvmInfo.isSimpleType(valueType)) {
+					value = props.getSimple(valueType.getTypeName(), keyString);
+				} else {
+					boolean valuePresent = props.getBoolean(keyString);
+					if (valuePresent) {
+						if (valueType.isArray()) {
+							final int arrayLength = props.getInt(keyString + ARRAY_SIZE);
+							ArrayType<?> valueArrayType = (ArrayType<?>) valueType;
+							CompositeType elementType = getCompositeType(valueArrayType.getElementOpenType());
+							CompositeData valueArray[] = new CompositeData[arrayLength];
```

Comment:
The brackets should be with the type: `CompositeData[] valueArray = ...`.

Reply:
Okay.  Fixed a few other places too.